withspectrum/spectrum
# Pro communities

This is getting close - @mxstbr if you're up to start a code review that'd be awesome.

My next tasks are really just UI polish around the upsells and upgrade flow for communities. I might refactor some of the invoice creation logic, but everything is working.

If you want to test webhooks locally, you'll need to:
1. install `stripe-local` globally with npm
2. run `stripe-local --key STRIPE_TEST_KEY_ALUE --url http://localhost:3001/stripe`

**Also you'll need to run a migration locally. There are two new migrations that we'll have to run in Prod - I will be adding a third migration tomorrow to clean up the invoices records in the db**

What that library does is pings Stripe every 5 seconds for new event logs, then passes those events to the local api endpoint.

Some things to test:
1. Non pro communities shouldn't be able to create a private channel
2. Pro communities can create a private channel
3. The thread composer should be available/unavailable depending on if the community is pro - when viewing the channel view
4. Private channels should be hidden from the thread composer dropdown in *any* view if the community is not pro
5. Upgrading/downgrading a community + personal pro subscription should only create 2 recurring payments, 2 invoices, and one customer in Stripe.

I'll be giving this a thorough pass over a couple more times before feeling comfortable shipping, but wanted to get the code in front of you sooner than later :)
